app/services/oregon_digital/triple_powered_service.rb

```diff
@@ -0,0 +1,131 @@
+# frozen_string_literal: true
+
+require 'triplestore_adapter'
+
+module OregonDigital
+  # Triple Powered Service to fetch out data from uri and grouped as 'label$uri'
+  class TriplePoweredService
+    # rubocop:disable Metrics/AbcSize
+    # rubocop:disable Metrics/MethodLength
+    # METHOD: A check function to identify if date fetch need to happen or not and fetch the top label first
+    def check_and_fetch_label(uris, parse_date: false)
+      if parse_date
+        fetch_all_labels_with_date(uris).first
+      else
+        fetch_all_labels(uris).first
+      end
+    end
+
+    # METHOD: Functionality to fetch all labels and put it as 'label$uri'
+    def fetch_all_labels(uris)
+      labels = []
+      uris.each do |uri|
+        graph = fetch_from_store(uri)
```

Comment:
Ok, this one might be a difficult one to explain, so let me know if we need to jump on a call later.

I don't think we need the fetching part or `#predicate_labels` part of this service
Our controlled properties system is different from SA@OSU and already includes fetching. So we're able to do something like `work.subject` to give us an array of `OregonDigital::ControlledVocuabulary::Subject` objects, each with a `#fetch` method that will setup the object to have a readable label.
You should be able to get the `label$uri` setup with something like this:
```
subjects = object.subject
fetch_all_labels(subjects)

...

def fetch_all_labels(controlled_vocabs)
labels = []
controlled_vocabs.each do |cv|
  cv.fetch
  labels << "#{cv.rdf_label.first.to_s}$#{cv.rdf_uri.to_s}"
end
```

Reply:
I will give this a try, if anything i will reach out. But thank you for the suggestion to fix this. I will get on to it.